pkg/products/observability/reconciler.go

```diff
@@ -0,0 +1,369 @@
+	}
+
+	op, err := controllerutil.CreateOrUpdate(ctx, serverClient, oo, func() error {
+		disabled := true
+		oo.Spec = observability.ObservabilitySpec{
+			ConfigurationSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"monitoring-key": r.Config.GetLabelSelector(),
+				},
+				MatchExpressions: nil,
+			},
+			SelfContained: &observability.SelfContained{
+				DisableRepoSync:         &disabled,
+				DisableObservatorium:    &disabled,
+				DisablePagerDuty:        &disabled,
+				DisableDeadmansSnitch:   &disabled,
+				DisableBlackboxExporter: nil,
+				SelfSignedCerts:         nil,
+				FederatedMetrics:        nil,
+				PodMonitorLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				PodMonitorNamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				ServiceMonitorLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				ServiceMonitorNamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				RuleLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				RuleNamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				ProbeLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				ProbeNamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"monitoring-key": r.Config.GetLabelSelector(),
+					},
+				},
+				AlertManagerConfigSecret: alertManagerConfigSecretName,
+			},
+			ResyncPeriod: "1h",
+		}
+
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseInProgress, err
+	}
+	if op == controllerutil.OperationResultUpdated || op == controllerutil.OperationResultCreated {
+		return integreatlyv1alpha1.PhaseInProgress, nil
+		//	r.log.Info(string(integreatlyv1alpha1.PhaseCompleted))
+		//} else {
+		//	return integreatlyv1alpha1.PhaseInProgress, nil
+	}
+
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: "observability-stack", Namespace: productNamespace}, oo)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	if oo.Status.StageStatus == observability.ResultFailed {
+		return integreatlyv1alpha1.PhaseFailed, nil
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+
+}
+
+func (r *Reconciler) reconcileSubscription(ctx context.Context, serverClient k8sclient.Client, _ *integreatlyv1alpha1.RHMI, productNamespace string, operatorNamespace string) (integreatlyv1alpha1.StatusPhase, error) {
```

Comment:
```suggestion
func (r *Reconciler) reconcileSubscription(ctx context.Context, serverClient k8sclient.Client, productNamespace string, operatorNamespace string) (integreatlyv1alpha1.StatusPhase, error) {
```
Should we remove `RHMI` as a function parameter as it's unused anyway ? :thinking:

Reply:
Good catch, fixed in latest commit.